net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_guiconfig.inc

```diff
@@ -0,0 +1,338 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// pfSense includes
+require_once('util.inc');
+
+// WireGuard includes
+require_once('wireguard/wg_globals.inc');
+
+// Formats peer endpoint address and port for the UI
+function wg_format_endpoint($header = true, $peer = null, $endpoint_key = 'endpoint', $port_key = 'port') {
+
+	if ($header) {
+
+		$endpoint = gettext('Endpoint');
+
+		$port = gettext('Port');
+
+		return "{$endpoint} : {$port}";
+
+	}
+
+	if (!is_null($peer)) {
+
+		if (!empty($peer[$endpoint_key])) {
+
+			return "{$peer[$endpoint_key]}:{$peer[$port_key]}";
+
+		} else {
+
+			return gettext('Dynamic');
+
+		}
+
+	}
+
+}
+
+// Gets a list of available tunnels in a format suitable for input selection boxes
+function wg_get_tun_list() {
+	global $config, $wgg;
+
+	$a_ret = array();
+
+	// Always include the unassigned option first
+	$a_ret['unassigned'] = 'Unassigned';
+
+	if (is_array($wgg['tunnels'])) {
+
+		foreach ($wgg['tunnels'] as $tunnel) {
+
+			$a_ret[$tunnel['name']] = "{$tunnel['name']} ({$tunnel['descr']})";	
+
+		}
+	
+	}
+
+	return $a_ret;
+
+}
+
+// Truncates a given string if if necessary and formats it to make it look good
+function wg_truncate_pretty($str, $length) {
+
+	$s_truncated = substr($str, 0, $length);
+
+	$s_pretty_dots = (strlen($str) > $length) ? '...' : '';
+
+	return "{$s_truncated}{$s_pretty_dots}";
+
+}
+
+function wg_interface_status_icon($status, $fa_icon = "fa-ethernet") {
+	global $wgg;
+
+	$ret_class = ($status == "up") ? 'text-success' : '';
+
+	$ret_html = "<i class=\"fa {$fa_icon} {$ret_class}\" style=\"vertical-align: middle;\" title=\"{$status}\"></i>";
+
+	return $ret_html;
+
+}
+
+// Returns the appropriate fa icon class for handshake status icon
+function wg_handshake_status_icon($latest_handshake = 0, $fa_icon = "fa-handshake") {
+	global $wgg;
+
+	$current_time = time();
+
+	$diff_time = $current_time - $latest_handshake;
+
+	$a_thresholds = $wgg['handshake_thresholds'];
+
+	foreach ($a_thresholds as $threshold => $value) {
+
+		if ($diff_time >= $threshold) {
+	
+			$ret_class = $value['class'];
+
+			$ret_title = $value['title'];
+
+			break;
+
+		}
+
+	}
+
+	$ret_html = "<i class=\"fa {$fa_icon} {$ret_class}\" style=\"vertical-align: middle;\" title=\"{$ret_title}\"></i>";
+
+	return $ret_html;
+
+}
+
+// Formats a given time difference in a human-friendly way
+function wg_human_time_diff($start_time, $end_time = null, $from_epoch = false) {
```

Comment:
Any particular reason you are doing this manually rather than using PHP's DateTime objects/functions which do this kind of thing natively?

Additionally working with epoch timestamps will fail on 32-bit ARM platforms in certain cases, though since these are all current timestamps and not future dates that's not a huge concern now.

I know it's not exactly the same, but look at what we do for certificates, for example at https://github.com/pfsense/pfsense/blob/master/src/etc/inc/certs.inc#L705 (for formatting) and https://github.com/pfsense/pfsense/blob/master/src/etc/inc/certs.inc#L2008 (for difference between times), and the related functions nearby.